### PR TITLE
ASM-4683  Issue with scheduling job queue

### PR DESCRIPTION
**Pull request by logankimmel**
Per email from Tex, the OS_Collector is also a direct update,
we were trying to schedule this and it was failing.  Also, Idrac
and Lifecycle should be run together first before anything else

### CODE REVIEW

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb [hunk 1]
```diff
@@ -34,13 +35,10 @@ def create
     sleep 20
     pre = []
     main = []
-    post = []
     @firmwares.each do |firmware|
       Puppet.debug(firmware)
-      if firmware["component_id"].to_i == LC_ID
+      if firmware["component_id"].to_i == (LC_ID || IDRAC_ID)
```
Comment (gavin-scott) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: This doesn't work to match either LC_ID or IDRAC_ID, just checks whether it's equal to LC_ID. You would need something like `[LC_ID, IDRAC_ID].include?(firmware["component_id"].to_i)`

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb [hunk 2]
```diff
@@ -50,14 +48,7 @@ def create
...
     update(main)
-    if post.size > 0
-      Puppet.debug("IDRAC update required, installing last")
-      update(post)
-
-      # idrac restarts after the firmware is installed. Sleep for a minute to
-      # ensure that is underway before we check for LC ready below.
-      sleep(60)
-    end
+    sleep(60)
```
Comment (gavin-scott) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: Do we only need to do this if IDRAC_ID was installed? I think leaving in a comment about the rationale would be a good idea.

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb (hunk 1 above)
Comment (logankimmel) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: How do you figure?
```
irb(main):001:0> a = 1
=> 1
irb(main):002:0> b = 2
=> 2
irb(main):004:0> 1 == (a || b)
=> true
irb(main):005:0> 3 == (a || b)
=> false
```

**Comment by logankimmel** [comment]
@gavin-scott This has been updated.

**Comment by gavin-scott** [comment]
@logankimmel verified that this does indeed fix the issue where a job repeatedly can't be scheduled.

**Pull request merged**